rfcs/20230323-add-split-graph-op/README.md

```diff
@@ -0,0 +1,142 @@
+
+
+<div align=center><img src=images/graph-compute-time.png></div>
+<div align=center>pic1 compute time</div>
+
+<div align=center><img src=images/split-concat.png></div>
+<div align=center>pic2 split-concat</div>
+
+
+## Proposal
+
+The proposal is to add `Split` operation to the operation set of Graph API.
+
+### Operations
+
+`Split` operation splits one single input tensor (`data`) and generates multiple outputs (`output_i`) of the specific length along a scalar `axis` attribute. It produces multiple output tensors based on `num_splits` or `size_splits` attribute. The input and output tensors should have the same data type and support floating point data types (f32, bf16, and f16).
+
+
+#### Proposed Attributes
+
+`Split` operation supports three attributes: `axis`, `num_splits` and `size_splits`. Both `axis` and `num_splits` should be an interger value. `size_splits` is a 1-D Tensor (or list). The input data is split into `num_splits` or len(`size_splits`) elements. The shape of the i-th element has the same size as the value except along dimension axis where the size is determined by `num_splits` or `size_splits` attribute.
+
+Attribute Name | Description | Value Type |Supported Values | Required or Optional
+-- | -- | --| --|--
+`axis` | specifies which dimension to split along. |s64 |A s64 value in the range of [-r, r-1] where r = rank(data)  | Required
+`num_splits` | number of outputs into which the input tensor data will be split along axis dimension. |s64 |A s64 value in the range of[1, data.dims[axis]]  | Optional
+`size_splits` |  list of sizes for each output alone axis dims. |s64 |A s64 list containing positive values, none is default  | Optional
+
+@note: 
+- Either `num_splits` or `size_splits` should be provided. When `num_splits` is used, `size_splits` will be ignored.
+- The dimension of input tensor data shape along `axis` must be evenly divisible by `num_splits` attribute.
+- The sum of the values in `size_splits` must be equal to the data.dims[`axis`].
+
+
+#### Proposed Inputs and Outputs
+
+There will be 1 input tensor and multiple output tensors for the proposed `Split` operation.
+##### Inputs
+| Index | Argument Name | Required or Optional |
+| ----- | ------------- | -------------------- |
+| 0     | `data`        | Required             |
```

Comment:
We use `src` and `dst` to represent inputs and outputs according to oneDNN's naming convention.

Reply:
Done, thanks for your remind!